qiskit_experiments/hamiltonian_tomography/cr_hamiltonian_experiment.py

```diff
@@ -0,0 +1,299 @@
+"""
+
+from typing import Tuple, List, Optional, Union
+
+import numpy as np
+from qiskit import QuantumCircuit, pulse, circuit
+from qiskit.exceptions import QiskitError
+from qiskit.providers import Backend
+from qiskit.providers.options import Options
+from qiskit.utils import apply_prefix
+
+from qiskit_experiments import BaseExperiment
+from .cr_hamiltonian_analysis import CRHamiltonianAnalysis
+
+
+class CRHamiltonianTomography(BaseExperiment):
+    r"""Cross resonance Hamiltonian tomography experiment.
+
+    This experiment assumes the 2 qubit Hamiltonian in the form
+
+    .. math::
+
+        H = \frac{I \otimes A}{2} + \frac{Z \otimes B}{2}
+
+    where :math:`A` and :math:`B` are Pauli operator :math:`\in {X, Y, Z}`.
+
+    This experiment is performed by stretching a pulse duration of the cross resonance
+    and measuring the target qubit by projecting onto x, y, and z basis.
+    Control qubit is prepared in both ground and excited state and the experiment is
+    repeated for both control qubit states.
+    See [SarahSheldon2016]_ for more technical details.
+
+    In this experiment, user can override the cross resonance pulse schedule.
+    This schedule defaults to a single cross resonance pulse with the flat-topped
+    Gaussian envelope (:class:`~qiskit.pulse.GaussianSquare`).
+    This can be replaced with the echoed CR with following code:
+
+    .. code-block::
+
+        # define parameter object. this should be defined with name="duration".
+        duration = Parameter("duration")
+
+        with build(backend, default_alignment="sequential") as ecr_sched:
+            u_channel = control_channels(0, 1)[0]
+            play(GaussianSquare(duration, 0.1, 64, duration-4*64), u_channel)
+            x(0)
+            play(GaussianSquare(duration, -0.1, 64, duration-4*64), u_channel)
+            x(0)
+
+        # create experiment with ECR pulse schedule.
+        durations = np.asarray([300, 400, 500], dtype=int)
+
+        ham_tomo = CRHamiltonianTomography(
+            qubits=[0, 1],
+            durations=durations,
+            cr_gate_schedule=ecr_sched,
+            x_values=durations * 2 * backend.configuration().dt
+        )
+
+    Note that schedule registered to the ``CRHamiltonianTomography`` should contain at least
+    one parameter object with ``name="duration"`` otherwise it raises an error.
+    In above example, the effect of rising and falling edges are not considered in x values.
+
+    .. [SarahSheldon2016] https://arxiv.org/abs/1603.04821
+    """
+    __analysis_class__ = CRHamiltonianAnalysis
+
+    def __init__(
+        self,
+        qubits: Tuple[int, int],
+        durations: np.ndarray,
+        cr_gate_schedule: Optional[Union[pulse.Schedule, pulse.ScheduleBlock]] = None,
+        x_values: Optional[np.ndarray] = None,
+    ):
+        r"""Create new CR Hamiltonian tomography experiment.
+
+        Args:
+            qubits: The qubit on which to run tomography.
```

Comment:
Is there any particular order of control and target? I.e. what does `(2,3)` do compared to `(3,2)`?